### PR TITLE
 category component

### PR DESCRIPTION
**Pull request by lina-jamal**
make form for select category an difficulty an fetch api as it

### CODE REVIEW

File: quiz/src/Category.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import {CategoryData,DifficultyData} from './Data';
+
+function Category () {
+  const[cat,setCat]=useState('');
+  const[diff,setDiff]=useState('');
+  const [questions,setQuestions]=useState([]);
+  useEffect(()=>{
+    fetch(`https://opentdb.com/api.php?amount=10&category=${cat}&difficulty=${diff}`)
+    .then(res => res.json()).then(({results})=>setQuestions(results))
+  })
```
Comment (AlaaTaima) [inline quiz/src/Category.js]: **Good work at using `useEffect` & `useState` :+1:**  ,
 - but you should let it run only  when the `diff` or `cat` changes.
 - To detect what is happening try :` console.log(results)` , you will find that results will be 
    consoled in an infinite loop without changing the value of `diff` and `cat`.
**SO,** 
`You can tell React to skip applying an effect if certain values (cat, diff) haven’t changed between re-renders. To do so, pass an array as an optional second argument to useEffect: `

File: quiz/src/Category.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     </select>
+
+     <label htmlFor = 'difficulty'> select difficulty</label>
+     <select name = {diff}  onChange={(e)=> setDiff(e.target.value)}> 
+     {DifficultyData.map((dif,index)=><option key ={index} value ={dif} >{dif}</option>)}
+     </select>
+  
+     <button type= 'submit' onClick={()=>{
+           
+        }}>Go To Questions </button>
```
Comment (AlaaTaima) [inline quiz/src/Category.js]: I think here you haven't finished yet and by clicking you want to show the related questions to the user.

File: quiz/src/Category.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  })
+
+  return(
+    <>
+    <form className='form-container'>
+
+     <label htmlFor='category'>Select category: </label>
+     <select name = {cat} onChange={(e)=> setCat(e.target.value)}>
+     {CategoryData.map((cat) => <option key={cat.id} value={cat.id}>{cat.name}</option>)}
+     </select>
```
Comment (AlaaTaima) [inline quiz/src/Category.js]: **Here, you used `htmlFor` with `label` to specifie which form element a label is bound to.**
so, 
you should use the attribute `id` with the desired element,  which here is `select` then give the `id` the same value of `htmlFor`

File: quiz/src/Category.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+     <label htmlFor = 'difficulty'> select difficulty</label>
+     <select name = {diff}  onChange={(e)=> setDiff(e.target.value)}> 
+     {DifficultyData.map((dif,index)=><option key ={index} value ={dif} >{dif}</option>)}
+     </select>
```
Comment (AlaaTaima) [inline quiz/src/Category.js]: here also use `id` with `htmlFor` as I explained in the previous comment.

File: quiz/src/Category.js (hunk 1 above)
Comment (AlaaTaima) [inline quiz/src/Category.js]: and also here handle error by using `catch`

File: quiz/src/Category.js (hunk 2 above)
Comment (lina-jamal) [reply quiz/src/Category.js]: yes i havent finished ,this part should do

File: quiz/src/Category.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  const[cat,setCat]=useState('');
+  const[diff,setDiff]=useState('');
+  const [questions,setQuestions]=useState([]);
+  const[show,setShow]=useState(true);
+  useEffect(()=>{
+
+    if(cat||diff){
+    fetch(`https://opentdb.com/api.php?amount=10&category=${cat}&difficulty=${diff}`)
+    .then(res => res.json()).then(({results})=>setQuestions(results))
+    .catch(error => console.error(error))}
```
Comment (AlaaTaima) [inline quiz/src/Category.js]: you can handle error here by using `hooks - useState`

File: quiz/src/Category.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import {CategoryData,DifficultyData} from './Data';
+
+function Category () {
+  const[cat,setCat]=useState('');
+  const[diff,setDiff]=useState('');
+  const [questions,setQuestions]=useState([]);
+  const[show,setShow]=useState(true);
+  useEffect(()=>{
+
+    if(cat||diff){
```
Comment (AlaaTaima) [inline quiz/src/Category.js]: good work here at validating values, that will prevent purposeless requests :+1:

File: quiz/src/Category.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    fetch(`https://opentdb.com/api.php?amount=10&category=${cat}&difficulty=${diff}`)
+    .then(res => res.json()).then(({results})=>setQuestions(results))
+    .catch(error => console.error(error))}
+    else{
+      setQuestions([]);
+    }
```
Comment (AlaaTaima) [inline quiz/src/Category.js]: here there is no need to empty the array of questions , it is already empty.
**More explanation :** 

- `cat` & `diff` will be empty just at the first time " after the initial rendering " and at this time questions array is empty so there is no need for `else statement` .
- Then during rendering the returned array of questions, you should check if the array is empty or not and show a descriptive message such as " No questions for this category " if the array is empty.

**Review by AlaaTaima — APPROVED** [review]

**Review by AlaaTaima — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**